src/depool.rs

```diff
@@ -699,3 +766,162 @@ async fn encode_transfer_stake(dest: &str, amount: u64) -> Result<String, String
         "amount": amount
     })).await
 }
+
+async fn call_contract(
+    conf: Config,
+    wallet: &str,
+    depool: &str,
+    value: &str,
+	keys: &str,
+    body: &str,
+    answer_is_expected: bool
+) -> Result<(), String> {
+    if conf.no_wait_for_answer {
+        send_with_body(conf.clone(), wallet, depool, value, keys, body).await
+    } else {
+        call_contract_and_get_answer(conf.clone(), wallet, depool, value,
+            keys, body, answer_is_expected).await
+    }
+}
+
+async fn call_contract_and_get_answer(
+    conf: Config,
+    src_addr: &str,
+    dest_addr: &str,
+    value: &str,
+	keys: &str,
+    body: &str,
+    answer_is_expected: bool
+) -> Result<(), String> {
+    let ton = create_client_verbose(&conf)?;
+    let abi = load_abi(MSIG_ABI)?;
+    let start = now();
+    
+    let params = json!({
+        "dest": dest_addr,
+        "value": convert::convert_token(value)?,
+        "bounce": true,
+        "allBalance": false,
+        "payload": body,
+    }).to_string();
+
+    let msg = prepare_message(
+        ton.clone(),
+        src_addr,
+        abi.clone(),
+        "submitTransaction",
+        &params,
+        None,
+        Some(keys.to_owned()),
+    ).await?;
+
+    print_encoded_message(&msg);
+
+    println!("Multisig message processing... ");
+    let callback = |_| {
+        async move {}
+    };
+
+    let result = send_message(
+        ton.clone(),
+        ParamsOfSendMessage {
+            message: msg.message.clone(),
+            abi: Some(abi.clone()),
+            send_events: false,
+            ..Default::default()
+        },
+        callback,
+    ).await
+    .map_err(|e| format!("Failed: {:#}", e))?;
+
+    wait_for_transaction(
+        ton.clone(),
+        ParamsOfWaitForTransaction {
+            abi: Some(abi.clone()),
+            message: msg.message.clone(),
+            shard_block_id: result.shard_block_id,
+            send_events: true,
+            ..Default::default()
+        },
+        callback.clone(),
+    ).await
+    .map_err(|e| format!("Failed: {:#}", e))?;
+    
+    println!("\nMessage was successfully sent to the multisig, waiting for message to be sent to the depool...");
```

Comment:
Maybe it's too early to print something here.

Reply:
No, that's after sending message to msig